### ISSUE TITLE
BUG:dataframe.groupby('some_column').timedelta.sum() results wrong when timedelta contains NaT (pandas=1.3.0)

### ISSUE THREAD

**Issue by minientertainment** [pandas-dev/pandas#42659]
- [y] I have checked that this issue has not already been reported.

- [y] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

**Note**: Please read [this guide](https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports) detailing how to provide the necessary information for us to reproduce your bug.

#### Code Sample, a copy-pastable example
this is a tiny example
```python
tmp = pd.DataFrame({'id': ['1624477460271-3908654213', '1624477460271-3908654213'], 'dt':[np.nan, pd.to_timedelta('0:0:2')]})
tmp
Out[6]: 
                         id              dt
0  1624477460271-3908654213             NaT
1  1624477460271-3908654213 0 days 00:00:02
```
#### Problem description

dataframe.groupby('some_column').timedelta.sum() results wrong when timedelta contains NaT
```python
tmp.groupby('id').dt.sum()
Out[7]: 
id
1624477460271-3908654213   -106752 days +00:12:45.145224192
Name: dt, dtype: timedelta64[ns]
```

#### Expected Output
```python
tmp.dt.sum()
Out[8]: Timedelta('0 days 00:00:02')
```
#### Output of ``pd.show_versions()``

<details>

[pandas 1.3.0]

</details>

**Comment by phofl** [comment]
Please povide something which is copy and pasteable (see https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports)

also please format your code

**Comment by minientertainment** [comment]
> Please povide something which is copy and pasteable (see https://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports)
> 
> also please format your code

updated

**Comment by nicolaslegrand91** [comment]
I also encountered this problem, it seems quite a major bug. 

@phofl  Here is a full copy-pastable example :

```python
import pandas as pd

df = pd.DataFrame({
    'a': [1, 1, 2, 2],
    'b': [pd.Timedelta('1d'), pd.Timedelta('2d'), pd.Timedelta('3d'), pd.NaT],
})

print('\ndf:')
print(df)

print('\ndf.dtypes:')
print(df.dtypes)

print('\ndf.groupby(\'a\').b.sum():')
print(df.groupby('a').b.sum())
```

![326DD059-EB73-4C3A-9AE3-41E8A8AD75B1](https://user-images.githubusercontent.com/20555592/127364967-55d28059-1346-40d0-9d63-8f5d864d357f.png)


#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : c7f7443c1bad8262358114d5e88cd9c8a308e8aa
python           : 3.9.5.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 20.4.0
Version          : Darwin Kernel Version 20.4.0: Thu Apr 22 21:46:47 PDT 2021; root:xnu-7195.101.2~1/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : en_US.UTF-8
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 1.3.1
numpy            : 1.21.1
pytz             : 2021.1
dateutil         : 2.8.1
pip              : 21.2.1
setuptools       : 49.6.0
Cython           : 0.29.24
pytest           : 6.2.3
hypothesis       : None
sphinx           : 3.5.4
blosc            : None
feather          : None
xlsxwriter       : 1.4.4
lxml.etree       : 4.6.3
html5lib         : None
pymysql          : 1.0.2
psycopg2         : 2.9.1 (dt dec pq3 ext lo64)
jinja2           : 2.11.3
IPython          : 7.25.0
pandas_datareader: None
bs4              : 4.9.3
bottleneck       : 1.3.2
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.4.2
numexpr          : None
odfpy            : None
openpyxl         : 3.0.7
pandas_gbq       : None
pyarrow          : None
pyxlsb           : None
s3fs             : None
scipy            : 1.7.0
sqlalchemy       : 1.4.22
tables           : None
tabulate         : 0.8.9
xarray           : None
xlrd             : 2.0.1
xlwt             : None
numba            : None

</details>

**Comment by simonjayhawkins** [comment]
first bad commit: [fa8f68e9a788901bc6459acb228c7cc8412791d3] BUG/API: SeriesGroupBy reduction with numeric_only=True (#41342)

cc @jbrockmendel

**Comment by simonjayhawkins** [comment]
changing milestone to 1.3.5

**Comment by jreback** [comment]
good to fix but not necessary for 1.3.x

**Comment by jbrockmendel** [comment]
Best guess is a `is_datetimelike` arg in `group_add` is needed
